core/src/main/java/bio/terra/pearl/core/service/survey/SurveyParseUtils.java

```diff
@@ -131,6 +131,26 @@ public static SurveyQuestionDefinition unmarshalSurveyQuestion(
         return definition;
     }
 
+    private static Pattern surveyQuestionRegexPattern = Pattern.compile("\\{\\s*\\w+\\.[\\w.]+\\s*}");
+
+    public static Map<String, List<String>> findReferencedSurveyQuestions(Survey survey) {
+        return surveyQuestionRegexPattern.matcher(survey.getContent()).results().map(result -> {
+            String cleaned = result.group().substring(1, result.group().length() - 1).trim();
+
+            int periodIndex = cleaned.indexOf('.');
+
+            String surveyStableId = cleaned.substring(0, periodIndex);
+            String questionStableId = cleaned.substring(periodIndex + 1);
+
+            return Map.entry(surveyStableId, questionStableId);
+        }).filter(
+                entry -> !List.of("profile", "proxyProfile").contains(entry.getKey())
+        ).reduce(new HashMap<>(), (acc, entry) -> {
+            acc.computeIfAbsent(entry.getKey(), k -> new ArrayList<>()).add(entry.getValue());
+            return acc;
+        }, (a, b) -> a);
```

Comment:
this might be a complex and expensive enough operation that it's worth computing it when the survey is created and storing it in the DB.  That could help with debugging and making this more performant